core/domain/image_services.py

```diff
@@ -44,44 +37,37 @@ def get_image_dimensions(file_content):
     Returns:
         tuple(int). Returns height and width of the image.
     """
-    img = Image.open(io.BytesIO(file_content))
-    width, height = img.size
+    image = Image.open(io.BytesIO(file_content))
+    width, height = image.size
     return height, width
 
 
 def compress_image(image_content, scaling_factor):
     """Compresses the image by resizing the image with the scaling factor.
 
-    Note that if the image's dimensions, after the scaling factor is applied,
-    exceed 4000 then the scaling factor will be recomputed and applied such that
-    the larger dimension of the image does not exceed 4000 after resizing. This
-    is due to an implementation limitation. See https://goo.gl/TJCbmE for
-    context.
-
     Args:
         image_content: str. Content of the file to be compressed.
         scaling_factor: float. The number by which the dimensions of the image
-            will be scaled. This is expected to be greater than zero.
+            will be scaled. This is expected to be between 0 and 1
 
     Returns:
         str. Returns the content of the compressed image.
     """
     if not constants.DEV_MODE:
-        height, width = get_image_dimensions(image_content)
+        if scaling_factor > 1:
+            raise ValueError('Scaling factor should be less than 1.')
+        image = Image.open(io.BytesIO(image_content))
+
+        image_format = image.format
+        width, height = image.width, image.height
```

Comment:
I saw your other note. I think one thing you could do, in order to allow reuse but also preserve the existing API, is to add a new private function in this file that takes an Image object and that returns its dimensions. That way, both this function and get_image_dimensions() can make use of it, and you don't need to duplicate the logic. (You'll need to think carefully about naming in order to avoid confusion.)